engine/variableValue.cc

```diff
@@ -32,14 +32,17 @@ namespace minsky
 
   const VariableValue& VariableValue::operator=(minsky::TensorVal const& x)
   {
-    bool realloc=numElements()!=x.data.size();
-    if (dims()!=x.dims) dims(x.dims);
-    if (realloc) allocValue();
-    memcpy(&valRef(), &x.data[0], x.data.size()*sizeof(x.data[0]));
-    return *this;
+	 bool realloc;
+	 // For feature 47
+     if (numSparseElements==0) realloc=numDenseElements()!=x.data.size();  
+     else realloc=numSparseElements!=x.data.size();
+     if (dims()!=x.dims) dims(x.dims);
+     if (realloc) allocValue(); 
+     memcpy(&valRef(), &x.data[0], x.data.size()*sizeof(x.data[0]));
```

Comment:
You also need to copy the index vector.

Reply:
Okay, I thought that &valRef() only needed index.size(). Do I thus have to replace m_idx in 

          if (size_t(m_idx+index.size())<=ValueVector::stockVars.size())
             return ValueVector::stockVars[m_idx];  

by index[0]?

Reply:
No - vaRef() return the reference to the first data element.